lib/utils/errors.go

```diff
@@ -18,21 +18,34 @@ package utils
 
 import (
 	"errors"
+	"net"
 	"strings"
 	"syscall"
 
 	"github.com/gravitational/teleport/api/constants"
 	"github.com/gravitational/trace"
 )
 
-// IsUseOfClosedNetworkError returns true if the specified error
-// indicates the use of closed network connection
-// TODO(dmitri): replace in go1.16 with `errors.Is(err, net.ErrClosed)`
+// IsUseOfClosedNetworkError returns true if the specified error indicates the
+// use of a closed network connection. If the error is a trace.Aggregate, all
+// the errors must be a closed network error.
```

Comment:
I think this particular behaviour against `trace.Aggregate` will be surprising. `error.Is` is closer to `Any()` than `All()`.

Reply:
Of the other two "aggregate error" types that I could find, one will do the `Any` behavior while the other will do the check on whatever the first error is; the behavior as I coded this was guided by the fact that in a lot of places we use `IsUseOfClosedNetworkError` as a check for "should we log this or can we ignore it?", so requiring that all the errors are ignorable felt more appropriate. Should we move the aggregate check to `IsOKNetworkError` instead?

Reply:
I moved the check for `trace.Aggregate` to `IsOKNetworkError`, as that one is only used to decide whether or not to log errors, AFAICT.